exporter.go

```diff
@@ -50,23 +51,54 @@ func getEnvBool(key string, defaultValue bool) bool {
 	return b
 }
 
-func getEnvDuration(key string, defaultValue time.Duration) time.Duration {
+func getEnvPositiveDuration(key string, defaultValue time.Duration) positiveDuration {
 	value, ok := os.LookupEnv(key)
 	if !ok {
-		return defaultValue
+		return positiveDuration{defaultValue}
+	}
+
+	posDur, err := parsePositiveDuration(value)
+	if err != nil {
+		log.Fatalf("Environment variable value for %s must be a positive duration: %v", key, err)
+	}
+	return posDur
+}
+
+// positiveDuration is a wrapper of time.Duration to ensure only positive values are accepted
+type positiveDuration struct{ time.Duration }
+
+func (pd *positiveDuration) Set(s string) error {
+	dur, err := parsePositiveDuration(s)
+	if err != nil {
+		return err
 	}
-	d, err := time.ParseDuration(value)
+
+	pd.Duration = dur.Duration
+	return nil
+}
+
+func parsePositiveDuration(s string) (positiveDuration, error) {
+	dur, err := time.ParseDuration(s)
 	if err != nil {
-		log.Fatalf("Environment variable value for %s must be a duration: %v", key, err)
+		return positiveDuration{}, err
+	}
+	if dur < 0 {
+		return positiveDuration{}, fmt.Errorf("negative duration %v is not valid", dur)
 	}
-	return d
+	return positiveDuration{dur}, nil
+}
+
+func createPositiveDurationFlag(dur time.Duration, key, helper string) *positiveDuration {
```

Comment:
just noticed that flag.Var uses value for the defalut value as well as name and usage for key and helper respectively. better to update them as well.  I think it makes sense to change the order of the parameters as well, so that the order is the same as for flag.Uint and the other similar functions. 

also, it makes sense to use positiveDuration as the default value rather than time.Duration

```suggestion
func createPositiveDurationFlag(name string, value positiveDuration, usage string) *positiveDuration {
```